components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/definitions/OAS3Parser.java

```diff
@@ -1176,10 +1196,25 @@ private void updateOperationManagedInfo(SwaggerData.Resource resource, Operation
             authType = "Application";
         }
         operation.addExtension(APIConstants.SWAGGER_X_AUTH_TYPE, authType);
-        if (resource.getPolicy() != null) {
+        // handle x-throttling-tier extension
+        if (resource.getPolicy() != null && !StringUtils.isEmpty(resource.getPolicy())) {
             operation.addExtension(APIConstants.SWAGGER_X_THROTTLING_TIER, resource.getPolicy());
         } else {
-            operation.addExtension(APIConstants.SWAGGER_X_THROTTLING_TIER, APIConstants.DEFAULT_API_POLICY_UNLIMITED);
+            operation.addExtension(APIConstants.SWAGGER_X_THROTTLING_TIER,
+                    APIUtil.getThrottlingTierFromThrottlingLimit(resource.getThrottlingLimit()));
```

Comment:
Do we need a null check for `resource.getThrottlingLimit()`

Reply:
fixed with: https://github.com/wso2/carbon-apimgt/pull/11760/commits/e5916cf71524fd2dda951e046c65c7454b8faef4